client/lib/cart-values/index.js

```diff
@@ -167,27 +167,15 @@ function isPayPalExpressEnabled( cart ) {
 }
 
 function isNetherlandsIdealEnabled( cart ) {
-	return (
-		config.isEnabled( 'upgrades/netherlands-ideal' ) &&
-		cart.allowed_payment_methods.indexOf( 'WPCOM_Billing_Stripe_Source_Ideal' ) >= 0 &&
-		'EUR' === cart.currency
```

Comment:
Just to acknowledge that the removal of `'EUR' === cart.currency` is also handled on the backend.